src/main/webapp/app/localvc/repository-view/repository-view.component.ts

```diff
@@ -79,11 +81,11 @@ export class RepositoryViewComponent implements OnInit, OnDestroy {
             this.participationCouldNotBeFetched = false;
             const exerciseId = Number(params['exerciseId']);
             const participationId = Number(params['participationId']);
-            if (participationId) {
+            this.repositoryType = participationId ? 'USER' : params['repositoryType'];
```

Comment:
The way the routes are set up is that we will have:

For the special repositories, e.g.
```
https://artemis-test2.artemis.cit.tum.de/course-management/18/programming-exercises/31/repository/TEMPLATE
```

In which case `params['repositoryType']` will be `'TEMPLATE'`.

For student repositories:

```
https://artemis-test2.artemis.cit.tum.de/course-management/18/programming-exercises/31/participations/97/repository
```

In which case `participationId` will be 97 (i.e., a number >= 1), so params['repositoryType'] won't be accessed (it would be undefined here).

TL;DR: It can be undefined, but it won't be used in that case